device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -627,6 +627,20 @@ void startTwinInternal(IotHubEventCallback twinStatusCallback, Object twinStatus
         }
     }
 
+    // only used by the MultiplexingClient class to signal to this client that it needs to re-register twin
+    // callbacks
+    void markTwinAsUnsubscribed()
+    {
+        this.twin = null;
+    }
+
+    // only used by the MultiplexingClient class to signal to this client that it needs to re-register methods
+    // callbacks
+    void markMethodsAsUnsubscribed()
+    {
+        this.method = null;
```

Comment:
Elsewhere in this class, this field is assigned when the user wants to subscribe to methods. When not null, this layer assumes that it is already subscribed. So setting this value to null again allows users to subscribe to methods again later if they re-register